src/main/java/uk/gov/ons/ctp/integration/event/generator/endpoint/PubSubEndpoint.java

```diff
@@ -0,0 +1,146 @@
+    String subscriptionName = pubSub().createSubscription(eventType);
+    return ResponseEntity.ok(subscriptionName);
+  }
+
+  @RequestMapping(value = "/pubsub/flush/{subscriptionName}", method = RequestMethod.GET)
+  @ResponseStatus(value = HttpStatus.OK)
+  public ResponseEntity<HttpStatus> flushQueue(
+      @PathVariable(value = "subscriptionName") final String subscriptionName) throws Exception {
+    log.info("Flushing queue: '" + subscriptionName + "'");
+    pubSub().flushTopic(getEventType(subscriptionName));
+    return ResponseEntity.ok().build();
+  }
+
+  @RequestMapping(value = "/pubsub/get/{subscriptionName}", method = RequestMethod.GET)
+  @ResponseStatus(value = HttpStatus.OK)
+  public ResponseEntity<String> get(
+      @PathVariable(value = "subscriptionName") final String subscriptionName,
+      @RequestParam String timeout) throws Exception {
+
+    log.info("Getting from queue: '" + subscriptionName + "' with timeout of '" + timeout + "'");
+
+    String messageBody = pubSub().getMessage(subscriptionName,
+        TimeoutParser.parseTimeoutString(timeout));
+
+    if (messageBody == null) {
+      return ResponseEntity.notFound().build();
+    }
+
+    return ResponseEntity.ok(messageBody);
+  }
+
+  @RequestMapping(value = "/pubsub/get/object/{subscriptionName}", method = RequestMethod.GET)
+  @ResponseStatus(value = HttpStatus.OK)
+  public ResponseEntity<String> get(
+      @PathVariable(value = "subscriptionName") final String subscriptionName,
+      @RequestParam String clazzName,
+      @RequestParam String timeout)
+      throws Exception {
+
+    log.info(
+        "Getting from subscription: '"
+            + subscriptionName
+            + "' and converting to an object of type '"
+            + clazzName
+            + "', with timeout of '"
+            + timeout
+            + "'");
+
+    EventType eventType = getEventType(subscriptionName);
+    if (eventType == null)  {
+      return ResponseEntity.notFound().build();
+    }
+
+    // Read message as object
+    Class<?> clazz = Class.forName(clazzName);
+    Object resultAsObject =
+        pubSub().getMessage(eventType, clazz, TimeoutParser.parseTimeoutString(timeout));
+
+    // Bail out if no object read from queue.
+    if (resultAsObject == null) {
+      return ResponseEntity.notFound().build();
+    }
+
+    // Convert object to string
+    String messageBody = mapper.writeValueAsString(resultAsObject);
+
+    if (messageBody == null) {
+      return ResponseEntity.notFound().build();
+    }
+
+    return ResponseEntity.ok(messageBody);
+  }
+
+  @RequestMapping(value = "/pubsub/close", method = RequestMethod.GET)
+  @ResponseStatus(value = HttpStatus.OK)
+  public ResponseEntity<String> close() throws Exception {
+    log.info("Closing PubSub channel connection");
+    PubSubHelper.destroy();
+    return ResponseEntity.ok("Connection closed");
+  }
+
+  private PubSubHelper pubSub() throws CTPException {
```

Comment:
Minor point
Would have been slightly neater to make PubSubHelper a bean in `EventGeneratorApplication` and then it could have been injected. No need same logic each time.